doc/elements.md

```diff
@@ -709,6 +709,59 @@ Please let the PrairieLearn developers know if you need a language that is not o
 
 -----
 
+## `pl-code-output` element
+
+Displays the value of a variable or parameter in python representation.
+
+#### Sample Element 1
+
+![](elements/pl-code-output.png)
+
+`server.py`
+
+```python
+data['params']['variable'] = pl.to_json({ 'a': 1, 'b': 2, 'c': 3 })
+```
+
+---
+
+`question.html`
+
+```html
+<pl-code-output variable-name="variable"></pl-code-output>
+```
+
+#### Sample Element 2
+
+```html
+<pl-code-output variable-name="df" variable-type="dataframe"></pl-code-output>
+```
+
+![](elements/pl-code-output2.png)
+
+#### Customizations
+
+Attribute | Type | Default | Description
+--- | --- | --- | ---
+`variable-name` | string | — | The name of the key in `data['params']` to get a value from
+`variable-type` | string | `text` | The type of variable that should be shown.  Either `text` for a textual representation as defined by `repr(var)`, or `dataframe` for a pandas dataframe.
```

Comment:
Now that you extended `pl.to_json()` to handle DataFrames, I think that we can automatically determine the type, right? But I can see that we might want to either output `repr()` or the HTML table for a DataFrame, so we need an attribute to control this. Would it be a good idea to replace `type` with `formatted=“true”` to indicate that we want “formatted” output (the HTML table). The `formatted` attribute would default to false. This would also allow us to add special handling of numpy arrays (for example) in the future, where they could be rendered also as HTML tables.

I’m not set on the `formatted` name — this could also be `text` or `repr` or something else. The main idea is just that it shouldn’t be specific to DataFrames, and it’s not really about the variable type, but rather about how it’s formatted for display.

Reply:
I've set this to just `text`, defaulting to `False`.